Core/include/Acts/TrackFinding/CombinatorialKalmanFilterExtensions.hpp

```diff
@@ -0,0 +1,99 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/TrackFitting/KalmanFitter.hpp"
+#include "Acts/Utilities/Result.hpp"
+
+#include <vector>
+
+namespace Acts {
+
+/// expected max number of track states that are expected to be added by
+/// stateCandidateCreator
+/// @note if the number of states exceeds this number dynamic memory allocation will occur.
+///       the number is chosen to yield a container size of 64 bytes.
+static constexpr std::size_t s_maxBranchesPerSurface = 10;
```

Comment:
Since we are breaking the API, do you think it makes sense to move this into a CKF related namespace?

Reply:
We can move this to the CfkTypes namespace or even move this and the CfkTypes to the extensions.

Reply:
Just experimenting with this. The  s_maxBranchesPerSurface  can certainly be moved into a namespace. I said one can move everything into the CKF extensions. The disadvantage would be that the full type name becomes a template of the track container and is very long. Thus, aliases would be used where ever these types appear. These aliases makes it often tedious to search and/or understand the actual definition of a type.  Anyway I would not have a strong opinion.

Reply:
Yes I agree we should not put it behind one of the templated classes. Potentially `CkfTypes` makes sense but I don't have a strong preference. I just thought `s_maxBranchesPerSurface` is a bit generic for the `Acts` namespace.

Reply:
moved s_maxBranchesPerSurface into CkfTypes